blog/04-03-23-a-guide-to-building-decentralized-loan-dapp/a-guide-to-building-decentralized-loan-dapp.md

```diff
@@ -0,0 +1,524 @@
+            ),
+            "Error during Loan Registration"
+        );
+
+        loan[loanLength] = LoanDetails(
+            payable(msg.sender),
+            _name,
+            _borrowerHomeAddress,
+            _occupation,
+            _purpose,
+            _nationalId,
+            "review",
+            _amount,
+            _loanTenure,
+            false,
+            block.timestamp
+        );
+        loanLength++;
+    }
+
+    function getLoans(uint256 _index)
+        public
+        view
+        isAdmin(_index)
+        returns (
+            address payable,
+            string memory,
+            string memory,
+            string memory,
+            string memory,
+            string memory,
+            string memory,
+            uint256,
+            uint256,
+            bool,
+            uint256
+        )
+    {
+        LoanDetails storage loans = loan[_index];
+        return (
+            loans.borrowerAddress,
+            loans.name,
+            loans.borrowerHomeAddress,
+            loans.occupation,
+            loans.purpose,
+            loans.nationalId,
+            loans.verification,
+            loans.amount,
+            loans.loanTenure,
+            loans.isPaid,
+            loans.timestamp
+        );
+    }
+
+    function verifyApplicant(uint256 _index) public payable isAdmin(_index) {
+        LoanDetails storage singleLoan = loan[_index];
+        require(
+            IERC20Token(cUsdTokenAddress).transferFrom(
+                bankAddress,
+                singleLoan.borrowerAddress,
+                singleLoan.amount
+            ),
+            "Could not disburse funds"
+        );
+        singleLoan.verification = "verified";
+    }
+
+    function unVerifyApplicant(uint256 _index) public isAdmin(_index) {
+        loan[_index].verification = "unverified";
+    }
+
+    // function to redeem loans with a fixed interest
+    function reedem(uint256 _index) public payable isOwner(_index) {
+        LoanDetails storage singleLoan = loan[_index];
+        require(
+            IERC20Token(cUsdTokenAddress).transferFrom(
+                msg.sender,
+                bankAddress,
+                (singleLoan.amount + singleLoan.amount * 1 / 10)
+            ),
+            "Did not reedem loan"
+        );
+        singleLoan.isPaid = true;
+    }
+
+    function isUserAdmin(address _address) public view returns (bool) {
+        if (_address == bankAddress) {
+            return true;
+        }
+        return false;
+    }
+
+    function getLoanLength() public view returns (uint256) {
+        return loanLength;
+    }
+}
+```
+
+I am going to do something different in this tutorial by leaving the implementation of the frontend as a challenge. You can use whatever framework of your choice, you can also check out my previous tutorial to get some ideas. [Check It Out]((https://docs.celo.org/blog/tutorials/how-to-build-car-marketplace-dapp-with-react)).
```

Comment:
`Fix The Brackets here [Check It Out](https://docs.celo.org/blog/tutorials/how-to-build-car-marketplace-dapp-with-react). make it like this`